packages/react/src/number-field/input/NumberFieldInput.tsx

```diff
@@ -20,9 +21,57 @@ const NumberFieldInput = React.forwardRef(function NumberFieldInput(
 ) {
   const { render, className, ...otherProps } = props;
 
-  const { getInputProps, inputRef, state } = useNumberFieldRootContext();
+  const {
+    allowInputSyncRef,
+    autoFocus,
+    disabled,
+    formatOptionsRef,
+    getAllowedNonNumericKeys,
+    getStepAmount,
+    id,
+    incrementValue,
+    inputMode,
+    inputValue,
+    invalid,
+    max,
+    mergedRef,
+    min,
+    name,
+    readOnly,
+    required,
+    setValue,
+    state,
+    valueRef,
+    setInputValue,
+    locale,
+  } = useNumberFieldRootContext();
+
+  const { getInputProps } = useNumberFieldInput({
+    allowInputSyncRef,
+    autoFocus,
+    disabled,
+    formatOptionsRef,
+    getAllowedNonNumericKeys,
+    getStepAmount,
+    id,
+    incrementValue,
+    inputMode,
+    inputValue,
+    invalid,
+    max,
+    mergedRef,
+    min,
+    name,
+    readOnly,
+    required,
+    setValue,
+    valueRef,
+    setInputValue,
+    locale,
+    inputRef: forwardedRef,
+  });
```

Comment:
I think we can just keep the logic inside the components, as this inflates the bundle size by passing all the params (same with the other components)

Reply:
Makes sense about the buttons as they previously were shared in the root - I think it's ok then